### PR TITLE
[IMP] l10n_es_edi_facturae{_adm_centers}: add missing facturae fields 

### PR DESCRIPTION
**Pull request by alialfie**
FileReference and FileDate fields are added to the Factura-E export.

task-3864452

### CODE REVIEW

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 1]
```diff
@@ -253,6 +253,8 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
             totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
 
             invoice_line_values.update({
+                'FileReference': self._l10n_es_edi_facturae_get_filename().split('.')[0],
+                'FileDate': fields.Date.today(),
```
Comment (william-andre) [inline addons/l10n_es_edi_facturae/models/account_move.py:257]:
```suggestion
                'FileDate': fields.Date.context_today(self),
```

File: addons/l10n_es_edi_facturae/models/account_move.py [hunk 2]
```diff
@@ -253,6 +253,8 @@ def _l10n_es_edi_facturae_inv_lines_to_items(self, conversion_rate=None):
             totals['total_taxes_withheld'] += sum((abs(tax["tax_amount"]) for tax in taxes_withheld_computed))
 
             invoice_line_values.update({
+                'FileReference': self._l10n_es_edi_facturae_get_filename().split('.')[0],
```
Comment (william-andre) [inline addons/l10n_es_edi_facturae/models/account_move.py:256]: The length can only be 20 char

**Review by william-andre — COMMENTED** [review]
Am I the only one thinking that it is stupid to have these values per invoice line? :zany_face:

**Review by alialfie — COMMENTED** [review]
Definitely stupid. If I had a nickel for everything that made no sense in factura-e I would have like 10 nickels which aren't that much but it's still weird

File: addons/l10n_es_edi_facturae/models/account_move.py (hunk 2 above)
Comment (william-andre) [reply addons/l10n_es_edi_facturae/models/account_move.py:256]: Neither solution is good because the spec is garbage...

**Comment by alialfie** [comment]
@william-andre updated according to chkl's comment on the task. thanksss

File: addons/l10n_es_edi_facturae/tests/data/expected_in_invoice_document.xml [hunk 3]
```diff
@@ -134,6 +134,8 @@
       </InvoiceTotals>
       <Items>
         <InvoiceLine>
+          <FileReference>BILL_2023_01_0001_facturae_signed</FileReference>
```
Comment (william-andre) [inline addons/l10n_es_edi_facturae/tests/data/expected_in_invoice_document.xml:137]: Isn't this more than 20 char? :smile:

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**